central/version/postgres/store.go

```diff
@@ -70,31 +69,26 @@ func (s *storeImpl) retryableUpsert(ctx context.Context, obj *storage.Version) e
 		return sac.ErrResourceAccessDenied
 	}
 
-	conn, release, err := s.acquireConn(ctx, ops.Get, "Version")
-	if err != nil {
-		return err
-	}
-	defer release()
-
-	tx, ctx, err := conn.Begin(ctx)
+	tx, ctx, err := s.begin(ctx)
 	if err != nil {
 		return err
 	}
 
 	if _, err := tx.Exec(ctx, deleteStmt); err != nil {
+		if errTx := tx.Rollback(ctx); errTx != nil {
+			return errTx
```

Comment:
Should we also wrap errors here:
```suggestion
			return errors.Wrapf(errTx, "rolling back transaction due to: %v", err)
```

Please check also other places with `Rollback`.